### ISSUE TITLE
Remove DRM from Amazon Book

### ISSUE THREAD

**Issue by nerdboy4200** [apprenticeharper/DeDRM_tools#854]
Hi,

I am helping someone with using your tool and can't remove the DRM on a book that they bought. They are using the Amazon Kindle App (Version 1.20.0 (47032) - with the appropriate change made to prevent the KFX conversion) on their Mac (running 10.11.6). On June 11th - they bought 2 books and at the time - they downloaded and converted the first one (Calibre 3.8) that day (with no issues). A bit over a week ago  - they downloaded the second one - and found that it would NOT remove the DRM from it. The first thing that we tried was to re-download the book and try again - and that didn't happen. We then upgraded DeDRM to 6.6.3 and that made no difference.

We we going to de-register - and noticed that it had re-registered on June 11. We did a quick de-register and re-register and re-downloaded the book and that made no difference. One thing I noticed was that when we went to the keys section in DeDRM in Calibre - it mentioned that it could not find the key (which I though was unusual).

The next thing we did was go by the FAQ (well except for one step) - and do the whole de-register, delete app, delete storage tree, reinstall the app (used the download dmg of 1.20 that we have had for a LONG time) (as well as re-disable the KFX creator) - then re-register. This made no difference.

Looking at the tree for DeDRM - I saw the tools directory - and found the kindlekey.pyw tool. I ran it and saw this (cleaned up username/systemname):

XX-iMac:Kindle_for_Mac_and_PC XX$ python kindlekey.pyw
Found k4Mac kinf2011 file: /Users/XX/Library/Application
Support/Kindle/storage/.kinf2011
Couldn't decrypt file.
XX-iMac:Kindle_for_Mac_and_PC XX$

I would like to note that nothing had changed from June 10 to today.

**Comment by ElleKayEm** [comment]
Post a log:

https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Comment by nerdboy4200** [comment]
Sorry for the delay - here is the log - hope this helps:

`calibre Debug log
calibre 3.8  embedded-python: True is64bit: True
Darwin-15.6.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '15.6.0', 'Darwin Kernel Version 15.6.0: Thu Jun 21 20:07:40
PDT 2018; root:xnu-3248.73.11~1/RELEASE_X86_64')
Python 2.7.12
OSX: ('10.11.6', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 6, 3) && Obok
DeDRM (6, 5, 4)
calibre 3.8  embedded-python: True is64bit: True
Darwin-15.6.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '15.6.0', 'Darwin Kernel Version 15.6.0: Thu Jun 21 20:07:40
PDT 2018; root:xnu-3248.73.11~1/RELEASE_X86_64')
Python 2.7.12
OSX: ('10.11.6', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 6, 3) && Obok
DeDRM (6, 5, 4)
devicePixelRatio: 1.0
logicalDpi: 72.0 x 72.0
physicalDpi: 103.000001547 x 103.000001547
Using calibre Qt style: True
Starting up...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
Started up in 2.17 seconds with 527 books
DeDRM v6.6.3: Trying to decrypt B07P14MH4T_EBOK.azw
Failed to check for calibre update: [Errno 64] Host is down
Failed to check for plugin update: [Errno 64] Host is down
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Cold for the Bastards of Pizzofalcone
Got DSN key from database default_key_4
Found 16 keys to try after 0.2 seconds
Crypto Type is: 2
DeDRM v6.6.3: Failed to decrypt with error: No key found in 16 keys tried.
DeDRM v6.6.3: Looking for new default Kindle Key after 0.2 seconds
Found k4Mac kinf2011 file: /Users/XX/Library/Application
Support/Kindle/storage/.kinf2011
Couldn't decrypt file.
DeDRM v6.6.3: Ultimately failed to decrypt after 3.1 seconds. Read the
FAQs at Harper's repository:
https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
   File "site-packages/calibre/customize/ui.py", line 171, in
_run_filetype_plugins
   File "calibre_plugins.dedrm.__init__", line 624, in run
   File "calibre_plugins.dedrm.__init__", line 574, in KindleMobiDecrypt
DeDRMError: DeDRM v6.6.3: Ultimately failed to decrypt after 3.1
seconds. Read the FAQs at Harper's repository:
https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Added Cold for the Bastards of Pizzofalcone to db in: 0.1
Added 1 books in 4.4 seconds`

**Comment by ElleKayEm** [comment]
Hmm, calibre with DeDRM and Kindle for Mac are on the same machine and user?  You can read this copy of the book in the Kindle app?  You've made sure the Kindle app doesn't auto-update?

**Comment by nerdboy4200** [comment]
Some notes:
* Issue is fixed - will explain below
* I'm sure it didn't auto-update - it had been disabled and was after every time it was re-installed - especially since an auto-update would mean that I had to make the one program non-executable.

So your going to love this. I went to have them open the book today - and lo and behold - the Kindle App decided that it was not registered (hows that for weird). The book that was downloaded had a Exclamation point showing that the app could now not read it (which wasn't there when we last tried it).

When we re-registered it - instead of it showing with an ID of 5 in the registration name - it showed as 4 - which was what it appeared to have been before all of this. After that - re-downloaded - and was able to DeDRM it and convert.

I'm wondering if Amazon did something crazy on their end (now why wouldn't that surprise me) - and then fixed it.

With that - I am closing this out. Thank you for the help and suggestions.
